src/NServiceBus.NHibernate/SynchronizedStorage/INHibernateSynchronizedStorageSession.cs

```diff
@@ -1,9 +1,13 @@
 ﻿namespace NServiceBus
 {
+    using System;
+    using System.Threading.Tasks;
     using global::NHibernate;
 
     interface INHibernateSynchronizedStorageSession
     {
         ISession Session { get; }
+
+        void RegisterCommitHook(Func<Task> callback);
```

Comment:
@andreasohlund I don't think we should aim for alignment as the goal would be to have the same hook in native SQL persistence.  On the other hand, I am not attached to the current name so if you thing `OnCommit` is better, I am 👍 to change.